src/vscode-bicep/src/visualizer/app/components/Graph/style.ts

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import { Stylesheet } from "cytoscape";
+
+import moduleSvg from "../../assets/icons/graph/module.svg";
+
+const backgroundColor = "#333333";
+const backgroundColorSecondary = "#3f3f3f";
+const foregroundColor = "#ffffff";
+const foregroundColorSecondary = "#9c9c9c";
+const errorColor = "red";
```

Comment:
Is there a way to inherit colors from the user theme rather than hard-coding them? Do we need to think about accessibility here?

Reply:
Good question - technically we can read the colors from the user theme and send them to the webview, but there are concerns though:
1. A color theme usually contains numerous background and foreground color definitions for different UI areas. It will be hard to choose what colors to use.
2. We don't know what theme a user will use and if the graph will look good in that theme.

I'm leaning towards defining our own themes for the graph. This PR adds a dark theme, but later we can add a light theme and a high contrast theme (which addresses the accessibility issue).

Reply:
Is there no standardization for how the colors are used? My assumption is that colors used for background vs foreground would have enough contrast to let people tell them apart.

Reply:
Yes good call. There is [Web Content Accessibility Guidelines (WCAG)](https://www.w3.org/WAI/standards-guidelines/wcag/glance/). Basically I think we'll want to meet the level AA success contrast criteria for the dark/light themes, and level AAA for the high contrast theme. There's an online tool to test that: https://webaim.org/resources/contrastchecker/. 
I realized the `foregroundColorSecondary` is a bit too dark which does not pass the level AA test. I'll make some adjustment.

Reply:
But even beyond that. VS code's own controls must be mapping these colors to themes somehow. Or rather each color "identifier" in VS code should have an intended usage (background, or various flavors of foreground colors). If we reuse those colors and match the intent of the particular color, wouldn't we get something that is theme-agnostic?

 For example this tree view will always render ok no matter which theme you have (unless it's a really bad one):
![image](https://user-images.githubusercontent.com/22460039/117370591-76117c80-ae7b-11eb-90aa-5316e0440a15.png)

I could be missing something here, though.

Reply:
For a VSCode theme you need to define colors for each UI controls individually, so it's basically a 1:1 mapping. I think the challenge is, because the colors are not designed for rendering a graph, it's pretty hard to map them to our graph components. For example, what color we should use as the canvas background? What color should we use as the graph node background? `sideBar.background` or `panel.background`? In some themes, those background colors are different (such as the built-in dark theme), while in other themes they can be the same (like the one I use):

![image](https://user-images.githubusercontent.com/16367959/117373423-0baf0b00-ae80-11eb-9dc2-3c0922ec0203.png)

I'm not against to this approach as it avoids designing our own color scheme, and it provides a more consistent UX, but I doubt we can find a theme-agnostic mapping between theme colors and the graph components.